src/main/java/core/mate/academy/service/MachineProducerStrategy.java

```diff
@@ -0,0 +1,24 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Machine;
+import core.mate.academy.service.producerimpl.BulldozerProducerImpl;
+import core.mate.academy.service.producerimpl.ExcavatorProducerImpl;
+import core.mate.academy.service.producerimpl.TruckProducerImpl;
+import java.util.ArrayList;
+
+public class MachineProducerStrategy {
+    public MachineProducer<? extends Machine> getMachineProducerStrategy(
+            Class<? extends Machine> type) {
+        switch (type.getSimpleName()) {
```

Comment:
since you're operating on type, it's weird to use class name

why not comparing classes directly i.e. `type == Buldozer.class`?

Reply:
I can't use type in switch case. Ok, I changed it ti if else and used type==

Reply:
and you weren't suppose to use switch :)